server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -56,6 +56,7 @@ public class NRTReplicationEngine extends Engine {
     private final CompletionStatsCache completionStatsCache;
     private final LocalCheckpointTracker localCheckpointTracker;
     private final WriteOnlyTranslogManager translogManager;
+    private volatile boolean shouldCommit;
```

Comment:
I don't think this needs to be volatile because its not updated. it should be final and set once in the ctor.